PhysicsTools/PatAlgos/plugins/TauJetCorrFactorsProducer.cc

```diff
@@ -130,26 +126,24 @@ std::vector<JetCorrectorParameters> TauJetCorrFactorsProducer::params(
   return retVal;
 }
 
-float TauJetCorrFactorsProducer::evaluate(edm::View<reco::BaseTau>::const_iterator& tauJet,
-                                          std::shared_ptr<FactorizedJetCorrector>& corrector,
+float TauJetCorrFactorsProducer::evaluate(reco::BaseTau const& tauJet,
+                                          FactorizedJetCorrector& corrector,
                                           int corrLevel) {
-  corrector->setJetEta(tauJet->eta());
-  corrector->setJetPt(tauJet->pt());
-  corrector->setJetE(tauJet->energy());
-  return corrector->getSubCorrections()[corrLevel];
+  corrector.setJetEta(tauJet.eta());
+  corrector.setJetPt(tauJet.pt());
+  corrector.setJetE(tauJet.energy());
+  return corrector.getSubCorrections()[corrLevel];
 }
 
-void TauJetCorrFactorsProducer::produce(edm::Event& evt, const edm::EventSetup& es) {
+void TauJetCorrFactorsProducer::produce(edm::Event& evt, const edm::EventSetup& eventSetup) {
   // get tau-jet collection from the event
-  edm::Handle<edm::View<reco::BaseTau> > tauJets;
-  evt.getByToken(srcToken_, tauJets);
+  auto tauJets = evt.getHandle(srcToken_);
```

Comment:
Okay right, but this commit is outdated now because the esConsumes migration of the `TauJetCorrFactorsProducer` was done in https://github.com/cms-sw/cmssw/pull/34997.